edx_reverification_block/xblock/reverification_block.py

```diff
@@ -231,6 +249,7 @@ def studio_submit(self, data, suffix=''):
         """
         related_assessment = data.get('related_assessment')
         attempts = data.get('attempts')
+        grace_period = data.get('grace_period')
```

Comment:
Yes it look alike it is doing conversion automatically. Reason is attempts are integer and no conversion done for that either.